src/main/src/main/java/org/geoserver/security/SecuredResourceNameChangeListener.java

```diff
@@ -0,0 +1,170 @@
+                            event.getOldValues().get(event.getPropertyNames().indexOf("name")));
+            String newWorkspaceName =
+                    String.valueOf(
+                            event.getNewValues().get(event.getPropertyNames().indexOf("name")));
+            // dont go anything if name has not changed
+            if (oldWorkspaceName.equalsIgnoreCase(newWorkspaceName)) return;
+
+            // if a security rule exists for the workspaec whose name has been changed
+            if (workspaceHasSecurityRule(oldWorkspaceName)) {
+                LOGGER.info(
+                        String.format(
+                                "Updating Security Rules for Renamed Workspace: %s",
+                                oldWorkspaceName));
+
+                // get rules in which workspace name should be updated
+                List<DataAccessRule> rulesToModifyList =
+                        getDataAccessRule(event.getSource(), oldWorkspaceName);
+                for (DataAccessRule rule : rulesToModifyList) rule.setRoot(newWorkspaceName);
+
+                try {
+                    dao.storeRules();
+                } catch (IOException e) {
+                    LOGGER.log(Level.SEVERE, e.getMessage(), e);
+                }
+            }
+        } else if (event.getSource() instanceof ResourceInfo) {
+            // if a layer has been renamed
+            String oldResourceName =
+                    String.valueOf(
+                            event.getOldValues().get(event.getPropertyNames().indexOf("name")));
+            String newResourceName =
+                    String.valueOf(
+                            event.getNewValues().get(event.getPropertyNames().indexOf("name")));
+            // dont do anything if name has not changed
+            if (oldResourceName.equalsIgnoreCase(newResourceName)) return;
+            // similar layer names can exist in different workspaces
+            ResourceInfo resourceInfo = (ResourceInfo) event.getSource();
+            String workspaceName = resourceInfo.getStore().getWorkspace().getName();
+            // if a security rule exists for the layer whose name has been changed
+            if (layerHasSecurityRule(workspaceName, oldResourceName)) {
+                LOGGER.info(
+                        String.format(
+                                "Updating Security Rules for Renamed Feature Type: %s",
+                                oldResourceName));
+
+                // get rules in which workspace name should be updated
+                List<DataAccessRule> rulesToModifyList =
+                        getDataAccessRule(resourceInfo, oldResourceName);
+                for (DataAccessRule rule : rulesToModifyList) rule.setLayer(newResourceName);
+
+                try {
+                    dao.storeRules();
+                } catch (IOException e) {
+                    LOGGER.log(Level.SEVERE, e.getMessage(), e);
+                }
+            }
```

Comment:
The issue also affects layer groups (both global and local), and they aren't handled here - please add them.